File: 05-binarytreepractice-Java/BinaryTree.java

class Node {
	public int value;
	public Node left, right;
	public Node(int value) {
		this.value = value;
		this.left = null;
		this.right = null;
	}
}

public class BinaryTree {
	public Node root;
	public static boolean flag = false;  

	public BinaryTree(int value) {
		this.root = new Node(value);
	}

	public boolean search(int value) {
		return search_Node(this.root, value);
	}

	private boolean search_Node(Node temp, int value) {
		if(root == null){  
		     flag = false;
		  }  
		  else{  
			//If value is found in the given binary tree then, set the flag to true  
			if(temp.value == value){  
			  flag = true;   
			}  
			//Search in left subtree  
			if(flag == false && temp.left != null){  
			   search_Node(temp.left, value);  
			}  
			//Search in right subtree  
			if(flag == false && temp.right != null){  
			   search_Node(temp.right, value);  
			}  

		  }
		  return flag;  
	}
	public static void main(String[] args){
		BinaryTree s = new BinaryTree(1);
		s.root.left = new Node(2);
		s.root.right = new Node(3);
		s.root.left.left = new Node(4);
		s.root.left.right = new Node(5);
		System.out.println(s.search(2));
	}
}

File: 05-binarytreepractice-Python/binarytree.py
class Node(object):
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

class BinaryTree(object):
    def __init__(self, root):
        self.root = Node(root)

    def search(self, find_val):
        if find_val < self.root.value:
            if self.left is None:
                return False
            return self.left.serach(find_val)
        elif find_val > self.root.value:
            if self.right is None:
                return False
            return self.right.findval(find_val)
        else:
            return self.root.value

    def print_tree(self):
        """Print out all tree nodes
        as they are visited in
        a pre-order traversal."""
        # Your code goes here
        pass

    def preorder_search(self, start, find_val):
        """Helper method - use this to create a 
        recursive search solution."""
        # Your code goes here
        pass

    def preorder_print(self, start, traversal):
        """Helper method - use this to create a 
        recursive print solution."""
        # Your code goes here
        pass

    tree = BinaryTree(1)
    tree.root.left = Node(2)
    tree.root.right = Node(3)
    tree.root.left.left = Node(4)
    tree.root.left.right = Node(5)
    print(tree.search(4))